### PR TITLE
aspnetcore-upload: Chunk Demo Implementation Completed

### PR DESCRIPTION
**Pull request by LanceMcCarthy**
## Summary

Two Items of consequence are in this Pull Request.

1. Updated project to .NET 6 (Microsoft ended support for NET 5 on May, 10th 2022)
2. Finished the implementation of the `ChunkUploadController` to support both concurrent and non-concurrent chunk uploads.

### Walkthrough

Here is what occurs in the lifecycle of the ChunkUpload ActionResult. To begin this lifecycle, the user needs to first select a large-ish file in the Chunk Upload view of the SPA app.

1. The [ChunkSave](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L49) controller ActionResult method is invoked and we save the chunk using the FileName and ChunkIndex to a temp file ([link to code](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L86-L94))
2. After each chunk upload, invoke `CombineChunks` and pass it the ChunkMetaData value to:
    * **Phase 1** - Verifying all chunks are present ([link to code](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L133-L147))
      * If they are not all present, we're done... escape method.
      * If all the chunks have been saved, move on to Phase 2
    * **Phase 2** - Combine the temp chunk files in the correct order using `metadata.ChunkIndex` ([link to code](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L149-L174))
    * **Phase 3** - Finalizing, saving final destination file ([link to code](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L176-L188))
    * **Phase 4** - Clean up temporary chunk files ([link to code](https://github.com/LanceMcCarthy/kendo-angular/blob/d48de4258501d8e10db74f84490b5e288388ac1b/examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs#L192-L210))

### Verification Testing

Here is a screenshot at runtime:

image

I have tested with both small and large files, as well as multiple files.

### Side Note

The SPA app and Kendo could use an update, but that's outside the scope of this Pull Request's intention. However, please note that even though there were no changes to the Kendo project, there was a modification to package.json.lock due to dependency updates. I recommend merging this with the AS.NET Core changes.

### CODE REVIEW

**Review by tsvetomir — DISMISSED** [review]
Great to have the interesting bits implemented 🔧  Big thanks!

File: examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs [hunk 1]
```diff
@@ -1,109 +1,108 @@
...
+            var metadata = serializer.ReadObject(ms) as ChunkMetaData;
+
             // The Name of the Upload component is "files"
             if (files != null)
             {
                 foreach (var file in files)
                 {
-                    //path = Path.Combine(_webHhostingEnvironment.WebRootPath, "Upload_Directory", somemetaData.FileName);
+                    // ************************************************************************** //
+                    // ********** Scenario 1 - Supports Only Concurrent Chunk Upload ************ //
```
[the remark this answers is not in the document]
Comment (LanceMcCarthy) [reply examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs:67]: Yeah, bad description in comment, will update.

[the remark this answers is not in the document]
Comment (LanceMcCarthy) [reply examples-standalone/aspnetcore-upload/Controllers/ChunkUploadController.cs]: Updated scenario descriptions to not only fix the swapped issue, but I also clarified it a little further to be easier for first-time readers.

**Comment by tsvetomir** [comment]
Totally forgot about this, sorry Lance! 🏖️

**Review by tsvetomir — APPROVED** [review]

**Comment by LanceMcCarthy** [comment]
Need to pull in a couple of commits form origin, and updated the Microsoft Extensions page.

I retested and everything still operates the same:
image

**Review by tsvetomir — APPROVED** [review]

**Review by svetq — APPROVED** [review]
The chunk upload does work as expected :rocket:

**Review by Raisolution — APPROVED** [review]

**Pull request merged**